### ISSUE TITLE
fan constantly turn on and off each 5 seconds for no reason

### ISSUE THREAD

**Issue by charliergi** [JackHack96/dell-xps-9570-ubuntu-respin#37]
I don't know why, but the fan is constantly turning on for 5 seconds then turn off for 10 seconds.

**Comment by kingjr** [comment]
Same here. This is particularly true when little is going on (e.g. libre office only)

**Comment by sjabasti** [comment]
Is it doing this only on AC or also on battery?

**Comment by charliergi** [comment]
It's doing this on both for me

**Comment by sjabasti** [comment]
Is tlp running ?
You also might want to take a look at https://github.com/JackHack96/dell-xps-9570-ubuntu-respin/wiki/Some-useful-tips-and-tricks#thermal-modes , and play with the thermal modes, i do not know if it will solve your problem

**Comment by kylescousin** [comment]
I have the same problem but on Linux Mint.

I wanted to switch to kubuntu just so I could use this script to make it work better with Linux.

Is this script not fixing the fan problems with this machine?

**Comment by JackHack96** [comment]
It seems a common problem, please try following @sjabasti advice

**Comment by kylescousin** [comment]
The issue has been resolved for me by upgrading to the latest Linux kernel 4.18.12

Use 'Ukuu'

**Comment by kylescousin** [comment]
@sjabasti Correct, I was too quick to cheer. The problem is still equally annoying.
Is there any other way to tone this thing down?

**Comment by sjabasti** [comment]
@kylescousin, Same question as earlier in this thread, is this an issue both on AC and on BAT (after a while)? Is TLP running? I am asking these questions as I had the same issue but only when on AC not on battery, and solved it by forcing TLP to run using the battery configuration until i have time to figure out which settings causes it.

**Comment by kylescousin** [comment]
@sjabasti I will try.

I'm also going to flash bios latest update 1.5.0 which released 4 days ago only.
*fingers crossed*

**Comment by JackHack96** [comment]
BIOS v1.5 is helping with Linux power management. I advice everyone to update

**Comment by kylescousin** [comment]
@JackHack96 So far I'm having a good experience after this bios upgrade, I hope it lasts. The fan is very silently running, but not bothersome.
At least the ON/OFF every 10 seconds stopped.

**Comment by charliergi** [comment]
For me I still have the ON/OFF probem with the 1.5.0 bios
Here is my tlp stats 
![image](https://user-images.githubusercontent.com/17712727/46723108-54c26280-cc77-11e8-85a3-ec9f150295f1.png)
And here is my grub configuration in case :

![image](https://user-images.githubusercontent.com/17712727/46723179-80dde380-cc77-11e8-8c88-bc36f51d0eb9.png)

**Comment by charliergi** [comment]
@sjabasti and the thermal modes are not doing anything, or do I have to reboot maybe ?

**Comment by charliergi** [comment]
@kylescousin Can you post your grub configuration located on /etc/default/grub and your tlp stats with 
"tlp stat -s" please ? Thanks !

**Comment by kylescousin** [comment]
@charliergi My laptop is running for 4 hours now and the fan hasn't gone mental yet... I do have almost 20 programs running. I'm on Linux Mint and using linux kernel 4.15.

I haven't tested 4.18 yet since the bios upgrade

```
# If you change this file, run 'update-grub' afterwards to update
# /boot/grub/grub.cfg.
# For full documentation of the options in this file, see:
#   info -f grub -n 'Simple configuration'

GRUB_DEFAULT=0
GRUB_TIMEOUT_STYLE=hidden
GRUB_TIMEOUT=10
GRUB_DISTRIBUTOR=`lsb_release -i -s 2> /dev/null || echo Debian`
GRUB_CMDLINE_LINUX_DEFAULT="quiet splash"
GRUB_CMDLINE_LINUX=""


# Uncomment to enable BadRAM filtering, modify to suit your needs
# This works with Linux (no patch required) and with any kernel that obtains
# the memory map information from GRUB (GNU Mach, kernel of FreeBSD ...)
#GRUB_BADRAM="0x01234567,0xfefefefe,0x89abcdef,0xefefefef"

# Uncomment to disable graphical terminal (grub-pc only)
#GRUB_TERMINAL=console

# The resolution used on graphical terminal
# note that you can use only modes which your graphic card supports via VBE
# you can see them in real GRUB with the command `vbeinfo'
#GRUB_GFXMODE=640x480

# Uncomment if you don't want GRUB to pass "root=UUID=xxx" parameter to Linux
#GRUB_DISABLE_LINUX_UUID=true

# Uncomment to disable generation of recovery mode menu entries
#GRUB_DISABLE_RECOVERY="true"

# Uncomment to get a beep at grub start
#GRUB_INIT_TUNE="480 440 1"

```



```
--- TLP 1.1 --------------------------------------------

+++ System Info
System         = Dell Inc. XPS 15 9570
BIOS           = 1.5.0
Release        = Linux Mint 19 Tara
Kernel         = 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018 x86_64
/proc/cmdline  = BOOT_IMAGE=/vmlinuz-4.15.0-36-generic root=/dev/mapper/mint--vg-root ro quiet splash vt.handoff=1
Init system    = systemd v237
Boot mode      = UEFI

+++ TLP Status
State          = enabled
Last run       = unknown
Mode           = unknown
Power source   = AC
```

**Comment by charliergi** [comment]
@kylescousin Thank you ! And what about battery life with your configuration ?

**Comment by randyzwitch** [comment]
The newest BIOS update seems to have helped me as well.

**Comment by satyarthap** [comment]
Hi Jack, Thanks for the re-spin image. Is the solution for this problem Bios Update? I'm currently on Bios 1.8.1 and the problem persists. A low fan starts for a few seconds then turns off, this cycle repeats forever.

Any help is appreciated.

**Comment by JackHack96** [comment]
If you are on AC it's normal that fan spin on and off in a cycle, because that's the way Dell programmed it in the BIOS. On battery, you should have silent fan. Check if TLP is running in battery mode :)
